File: angular-movieFrontend/src/Data/Response/GameResponse.ts
import { GenreResponse } from "./GenreResponse";
import { MediaStatsResponse } from "./MediaStatsResponse";
import { ReviewResponse } from "./ReviewResponse";

export interface GameResponse {
  id: string;
  title: string;
  description: string;
  genreResponse: GenreResponse;
  releaseDate: string; 
  language: string;
  reviews: ReviewResponse[] | null;
  mediaStats: MediaStatsResponse;
  developer: string;
  engine: string | null;
  pegiRating: number;
  supportsCrossPlay: boolean;
  platforms: string[];
}

File: Application/Features/Medias/Games/Common/GameResponse.cs
using Application.Features.Common.Interfaces;
using Application.Features.Common.Models;
using Application.Features.Genres.GetGenres;
using Application.Features.Medias.Movies.Common;
using Application.Features.UserInteractions.Reviews.Common;

namespace Application.Features.Medias.Games.Common
{
    public record GameResponse(
        Guid id,
        string Title,
        string Description,
        GenreResponse GenreResponse,
        DateTime ReleaseDate,
        string Language,
        List<ReviewResponse>? Reviews,
        MediaStatsResponse MediaStats,
        string Developer,
        string? Engine,
        int PegiRating,
        bool SupportsCrossPlay,
        List<string> Platforms
        ) : MediaResponse(id, Title, Description, GenreResponse, ReleaseDate, Language, MediaStats), IResponse;
}
